dashboard/pages/Dashboard.py

```diff
@@ -27,29 +27,124 @@ def get_db_connection() -> extensions.connection:
         return None
 
 
-def load_all_data(db_connection: extensions.connection) -> pd.DataFrame:
-    """Loads all the data from the database into a pandas dataframe"""
-    with db_connection.cursor() as curr:
-        curr.execute("""
+@st.cache_data
```

Comment:
General comment, Love that you've separated everything into separate SQL scripts and reduced the amount of pandas stuff. Probably makes the dashboard run faster